ext/reflection/php_reflection.c

```diff
@@ -2105,7 +2127,127 @@ ZEND_METHOD(ReflectionFunctionAbstract, getNumberOfRequiredParameters)
 }
 /* }}} */
 
-/* {{{ Returns an array of parameter objects for this function */
+/* {{{ Returns whether a parameter exists or not */
+ZEND_METHOD(ReflectionFunctionAbstract, hasParameter)
+{
+    reflection_object *intern;
+    zend_function *fptr;
+    zend_string *arg_name = NULL;
+    zend_long position;
+    uint32_t num_args;
+
+    ZEND_PARSE_PARAMETERS_START(1, 1)
+        Z_PARAM_STR_OR_LONG(arg_name, position)
+    ZEND_PARSE_PARAMETERS_END();
+
+    GET_REFLECTION_OBJECT_PTR(fptr);
+
+    num_args = fptr->common.num_args;
+
+    if (fptr->common.fn_flags & ZEND_ACC_VARIADIC) {
+        num_args++;
+    }
+
+    if (!num_args) {
+        RETURN_FALSE;
+    }
+
+    if (arg_name != NULL) {
+        if (ZSTR_LEN(arg_name) == 0) {
+            zend_argument_value_error(1, "must not be an empty string");
+            RETURN_THROWS();
+        }
+
+        if (get_parameter_position(fptr, arg_name, num_args) > -1) {
+            RETURN_TRUE;
+        }
+
+        RETURN_FALSE;
+    } else {
+        if (position < 0) {
+            zend_argument_value_error(1, "must be greater than or equal to 0");
+            RETURN_THROWS();
+        }
+
+        RETURN_BOOL(position < num_args);
+    }
+}
+/* }}} */
+
+/* {{{ Returns a parameter specified by its name */
+ZEND_METHOD(ReflectionFunctionAbstract, getParameter)
+{
+    reflection_object *intern;
+    zend_function *fptr;
+    zend_string *arg_name = NULL;
+    zend_long position;
+    struct _zend_arg_info *arg_info;
+    uint32_t num_args;
+    zval reflection;
+
+    ZEND_PARSE_PARAMETERS_START(1, 1)
+        Z_PARAM_STR_OR_LONG(arg_name, position)
+    ZEND_PARSE_PARAMETERS_END();
+
+    GET_REFLECTION_OBJECT_PTR(fptr);
+
+    num_args = fptr->common.num_args;
+    arg_info = fptr->common.arg_info;
+
+    if (fptr->common.fn_flags & ZEND_ACC_VARIADIC) {
+        num_args++;
+    }
+
+    if (num_args < 1) {
+        char *message;
+
+        if (fptr->common.scope) {
+            zend_spprintf(&message, 0, "Method %s::%s() has no parameters", ZSTR_VAL(fptr->common.scope->name), ZSTR_VAL(fptr->common.function_name));
+        } else {
+            zend_spprintf(&message, 0, "Function %s() has no parameters", ZSTR_VAL(fptr->common.function_name));
+        }
+
+        _DO_THROW(message);
+        RETURN_THROWS();
+    }
```

Comment:
You are not freeing ``message`` here, but maybe what makes more sense is to create a new ``_DO_THROW_EX()`` macro which wraps the ``zend_throw_exception_ex()`` function but allows to pass in a format string like sprintf

Reply:
I think I may update the other exceptions I've added to be more informative, so instead of just:

> The parameter specified by its name could not be found

or

> The parameter specified by its offset could not be found

It's something like:

> Method %s::%s() does not have a parameter by the name %s

and

> Method %s::%s() does not have a parameter at position "%i"